tools/src/android-update-native-dependencies/updateGradleFiles.ts

```diff
@@ -0,0 +1,156 @@
+import logger from '../Logger';
+import { AndroidProjectDependenciesUpdates } from './types';
+import { addColorBasedOnSemverDiff, calculateSemverDiff } from './utils';
+
+function replaceVersionInGradleFile(
+  body: string,
+  {
+    oldVersion,
+    newVersion,
+    fullName,
+    group,
+    name,
+  }: {
+    oldVersion: string;
+    newVersion: string;
+    fullName: string;
+    name: string;
+    group: string;
+  }
+): string {
+  let modifiedBody = body;
+
+  const regexVersionVariable = new RegExp(`${fullName}:\\\${?(\\w+)}?`, 'ig');
+
+  // 'de.kevcodez:pubg-api-wrapper:$myVar'
+  // 'de.kevcodez:pubg-api-wrapper:${myVar}'
+  const versionWithVariableMatches = regexVersionVariable.exec(modifiedBody);
+  if (versionWithVariableMatches?.length === 2) {
+    const variableName = versionWithVariableMatches[1];
+
+    const regexVariableDefinition = new RegExp(
+      `(${variableName}(\\s+)?=(\\s+)?('|")${oldVersion}('|"))`,
+      'ig'
+    );
+
+    regexVariableDefinition
+      .exec(modifiedBody)
+      ?.filter((it) => it.includes(oldVersion))
+      .forEach((match) => {
+        modifiedBody = modifiedBody.replace(match, match.replace(oldVersion, newVersion));
+      });
+
+    // val PUBG_API_WRAPPER by extra("0.8.1")
+    // eslint-disable-next-line no-useless-escape
+    const regexKotlinValExtra = new RegExp(`${variableName}.+\(("|')${oldVersion}("|')\)`);
+    regexKotlinValExtra
+      .exec(modifiedBody)
+      ?.filter((it) => it.includes(oldVersion))
+      .forEach((match) => {
+        modifiedBody = modifiedBody.replace(match, match.replace(oldVersion, newVersion));
+      });
+  }
+
+  // compile 'de.kevcodez:pubg-api-wrapper:1.0.0'
+  const regexVersionInline = new RegExp(`${fullName}:${oldVersion}`, 'g');
+  regexVersionInline
+    .exec(modifiedBody)
+    ?.filter((it) => it.includes(`${fullName}:${oldVersion}`))
+    .forEach((match) => {
+      modifiedBody = modifiedBody.replace(match, `${fullName}:${newVersion}`);
+    });
+
+  // id 'com.github.ben-manes.versions' version "0.21.0"
+  // id("com.github.ben-manes.versions") version "0.22.0"
+  const regexPluginVersionWithPrefix = new RegExp(
+    `${group}("|')\\)?(\\s+)?version(\\s+)?("|')${oldVersion}("|')`
+  );
+  regexPluginVersionWithPrefix
+    .exec(modifiedBody)
+    ?.filter((it) => it.includes(oldVersion)) // filter out all groups not containing version
+    .forEach((match) => {
+      modifiedBody = modifiedBody.replace(match, match.replace(oldVersion, newVersion));
+    });
+
+  // compile group: 'de.kevcodez.pubg', name: 'pubg-api-wrapper', version: '0.8.1'
+  const regexDependencyWithVersionPrefix = new RegExp(
+    `${name}('|"),(\\s+)?version:(\\s+)('|")${oldVersion}('|")`
+  );
+  regexDependencyWithVersionPrefix
+    .exec(modifiedBody)
+    ?.filter((it) => it.includes(oldVersion))
+    .forEach((match) => {
+      modifiedBody = modifiedBody.replace(match, match.replace(oldVersion, newVersion));
+    });
+
+  return modifiedBody;
```

Comment:
I'll need to read this function more carefully when I'm back 😅 I feel like it's pretty overcomplicated and probably we could use some helpers for file transformations that we already have (e.g. versioning scripts). Maybe @lukmccall could help you when you both are at the office.

Reply:
We decided to keep as is after all